src/Components/test/E2ETest/ServerExecutionTests/ServerSideAppTest.cs

```diff
@@ -41,7 +40,7 @@ public void HasTitle()
         [Fact]
         public void HasHeading()
         {
-            Assert.Equal("Hello, world!", Browser.FindElement(By.TagName("h1")).Text);
+            Browser.Equal("Hello, world!", () => Browser.FindElement(By.CssSelector("h1#index")).Text);
```

Comment:
This assert is better because it waits for the element to appear, which is good as the issue could come from the page not being navigated from the previous location. Also uses a more specific way to find the element for that reason.

Reply:
As per other comment, I agree it's right to switch from `Assert` to `Browser` here, but I disagree that we need to change the CSS selector.